lib/compress/zbuff_compress.c

```diff
@@ -113,20 +112,18 @@ ZBUFF_CCtx* ZBUFF_createCCtx_advanced(ZSTD_customMem customMem)
     {
         zbc = (ZBUFF_CCtx*)calloc(1, sizeof(ZBUFF_CCtx));
         if (zbc==NULL) return NULL;
-        zbc->customAlloc = malloc;
-        zbc->customFree = free;
+        memcpy(&zbc->customMem, &defaultCustomMem, sizeof(ZSTD_customMem));
         zbc->zc = ZSTD_createCCtx();
         return zbc;
     }
 
     if (!customMem.customAlloc || !customMem.customFree)
         return NULL;
 
-    zbc = (ZBUFF_CCtx*)customMem.customAlloc(sizeof(ZBUFF_CCtx));
+    zbc = (ZBUFF_CCtx*)customMem.customAlloc(customMem.opaque, sizeof(ZBUFF_CCtx));
```

Comment:
This is required to make custom memory allocation compatible with zlib:
    alloc_func zalloc;  /* used to allocate the internal state */
    free_func  zfree;   /* used to free the internal state */
    voidpf     opaque;  /* private data object passed to zalloc and zfree */